src/pages/CreatePool/index.tsx

```diff
@@ -431,107 +436,120 @@ export default function CreatePool(): React.ReactElement {
-                        setTokenInputs([...tokenInputs])
-                        if (
-                          new Set(tokenInputs).size !== tokenInputs.length &&
-                          e.target.value
-                        ) {
-                          setTokenInputErrorMsg("Duplicate Tokens Not Allowed")
-                        } else {
-                          setTokenInputErrorMsg("")
-                        }
-                      }}
-                      onBlur={(e) => void onTokenInputBlur(e.target, index)}
-                      helperText={
-                        tokenInfo[index]?.name
-                          ? `${tokenInfo[index]?.name} (${
-                              tokenInfo[index]?.symbol
-                            }: ${String(tokenInfo[index]?.decimals)} decimals)`
-                          : " "
-                      }
-                      InputProps={{
-                        endAdornment: (
-                          <InputAdornment position="end">
-                            {index > 1 && (
-                              <IconButton
-                                onClick={() =>
-                                  setTokenInputs((prev) =>
-                                    prev.filter(
-                                      (value, tokenIndex) =>
-                                        index !== tokenIndex,
-                                    ),
-                                  )
-                                }
-                              >
-                                {inputLoading[index] ? (
+                    {tokenInputs.map((tokenInput, index) => (
+                      <Box
+                        key={`token-input-${index}`}
+                        flexBasis={`calc(50% - ${theme.spacing(1.5)})`}
+                      >
+                        <TextField
+                          autoComplete="off"
+                          value={tokenInputs[index]}
+                          label={`Token ${index}`}
+                          fullWidth
+                          color={tokenInfo[index]?.checkResult ?? "primary"}
+                          margin="normal"
+                          onChange={(e) => {
+                            tokenInputs[index] = e.target.value
+                            setTokenInputs([...tokenInputs])
+                            if (
+                              new Set(tokenInputs).size !==
+                                tokenInputs.length &&
+                              e.target.value
+                            ) {
+                              setTokenInputErrorMsg(
+                                "Duplicate Tokens Not Allowed",
+                              )
+                            } else {
+                              setTokenInputErrorMsg("")
+                            }
+                          }}
+                          onBlur={(e) => void onTokenInputBlur(e.target, index)}
```

Comment:
I think it would be good to use `isAddress` to check the valid address on `onChange`event  rather check address on `onBlur` event.

`import { isAddress } from "ethers/lib/utils"`

Reply:
yeah, because we need to fetch only `isAddress` is true.